src/main/java/org/apache/ibatis/executor/ReuseExecutor.java

```diff
@@ -95,7 +95,7 @@ private Statement prepareStatement(StatementHandler handler, Log statementLog) t
 
   private boolean hasStatementFor(String sql) {
     try {
-      return statementMap.keySet().contains(sql) && !statementMap.get(sql).getConnection().isClosed();
+      return statementMap.containsKey(sql) && !statementMap.get(sql).getConnection().isClosed();
```

Comment:
It would be more efficient to assign the result of `get()` to a local variable. i.e. One less node lookup.

```suggestion
      Statement statement = statementMap.get(sql);
      return statement != null && !statement.getConnection().isClosed();
```